lib/spack/spack/schema/config.py

```diff
@@ -38,8 +38,7 @@
                 'additionalProperties': False,
                 'properties': {
                     'tcl': {'type': 'string'},
-                    'lmod': {'type': 'string'},
-                    'dotkit': {'type': 'string'},
+                    'lmod': {'type': 'string'}
```

Comment:
This should stay in the schema, but print an error when configuration loads to tell people we no longer support dotkit.  We can make it a breaking schema error in some future release.

Reply:
I ended up writing a new custom validator that works for both JSON objects and arrays. One example is:

https://github.com/spack/spack/blob/c754074ed54595a091ffe3cb478658b473c50008/lib/spack/spack/schema/modules.py#L176-L182

It could be further generalized (for instance we could support different messages for different properties) but I think we could extend it later if need be. If `error` is true this will yield a validation error, otherwise it will just print a warning.

As an example, with `error` set to false:
```console
$ spack module tcl refresh -y
==> Warning: specifying a "dotkit" module root has no effect [support for dotkit module files has been dropped]
==> Regenerating tcl module files
```
while setting it to true:
```console
$ spack module tcl refresh -y
==> Error: /home/mculpo/PycharmProjects/spack/etc/spack/defaults/config.yaml:33: specifying a "dotkit" module root has no effect [support for dotkit module files has been dropped]
```
I might need to come up with a couple of tests for this validator.

Reply:
Added a unit test for the schema validator